native/src/main/java/io/ballerina/stdlib/crypto/PasswordUtils.java

```diff
@@ -89,6 +114,39 @@ public static Object validateWorkFactor(long workFactor) {
         }
         return null;
     }
+    
+    /**
+     * Validate if the provided PBKDF2 iterations is within acceptable bounds.
+     *
+     * @param iterations the iterations count to validate
+     * @return null if valid, error if invalid
+     */
+    public static Object validatePBKDF2Iterations(long iterations) {
+        if (iterations < MIN_PBKDF2_ITERATIONS) {
+            return CryptoUtils.createError(
+                String.format("Iterations must be at least %d", MIN_PBKDF2_ITERATIONS)
+            );
+        }
+        return null;
+    }
+    
+    /**
+     * Validate if the provided PBKDF2 algorithm is supported.
+     *
+     * @param algorithm the HMAC algorithm to validate
+     * @return null if valid, error if invalid
+     */
+    public static Object validatePBKDF2Algorithm(String algorithm) {
```

Comment:
```suggestion
    public static Object validatePbkdf2Algorithm(String algorithm) {
```
Let's update other places